apps/api/src/domain/models/playlist/commands/translate-playlist-name/translate-playlist-name.command-integration.spec.ts

```diff
@@ -0,0 +1,184 @@
+
+const existingPlaylist = getValidAggregateInstanceForTest(AggregateType.playlist).clone({
+    name: new MultilingualText({
+        items: [
+            new MultilingualTextItem({
+                text: 'original name of playlist',
+                role: MultilingualTextItemRole.original,
+                languageCode: LanguageCode.Chilcotin,
+            }),
+        ],
+    }),
+});
+
+const englishName = 'playlist name translated to English';
+
+const validPayload: TranslatePlaylistName = {
+    aggregateCompositeIdentifier: {
+        type: AggregateType.playlist,
+        id: existingPlaylist.id,
+    },
+    languageCode: LanguageCode.English,
+    text: englishName,
+};
+
+const validCommandFSA = {
+    type: commandType,
+    payload: validPayload,
+};
+
+const buildValidCommandFSA = (): FluxStandardAction<DTO<TranslatePlaylistName>> => validCommandFSA;
+
+describe(commandType, () => {
+    let app: INestApplication;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }).catch((error) => {
+                throw error;
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(commandAssertionDependencies, {
+                systemUserId: dummySystemUserId,
+                buildValidCommandFSA,
+                initialState: new DeluxeInMemoryStore({
+                    [AggregateType.playlist]: [existingPlaylist],
+                }).fetchFullSnapshotInLegacyFormat(),
+                checkStateOnSuccess: async ({
+                    aggregateCompositeIdentifier: { id: playlistId },
+                }: TranslatePlaylistName) => {
+                    const playlistSearchResult = await testRepositoryProvider
+                        .forResource(ResourceType.playlist)
+                        .fetchById(playlistId);
+
+                    expect(playlistSearchResult).not.toBe(NotFound);
+
+                    const playlist = playlistSearchResult as Playlist;
+
+                    const englishTextItem = playlist.name.items.find(
+                        ({ languageCode }) => languageCode === LanguageCode.English
+                    );
+
+                    expect(englishTextItem.text).toBe(englishName);
```

Comment:
Check that the event record is persisted as well.

Reply:
done